src/product-service/handlers/getProductsById.ts

```diff
@@ -0,0 +1,32 @@
+import products from '../data.json';
+import { APIGatewayProxyEvent } from 'aws-lambda';
+export const getProductsById = async (event: APIGatewayProxyEvent) => {
+  const productId = event.pathParameters?.productId;
+  if (!productId) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({ message: 'ProductId dosnt exist' })
+    };
+  }
+  const product = products.find(product => product.id === productId)
+  console.log(event)
+  if (!product) {
+    return {
+      statusCode: 404,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+      },
+      body: 'Product not found'
+    }
+  }
+
+  return {
```

Comment:
I think this looks like a pattern that is repeated over and over again

You can create a helper for it

`src/utils/lambda.ts`

const generateHttpResponse = <T>(statusCode: number, response: T): string => {...}

Reply:
Done